From f42c4207387ff04267a9cdcd8da0aed3ea3f05be Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 27 Dec 2020 11:44:57 +0100
Subject: [PATCH 1/1] drivers: qe: avoid double free()

Avoid calling free(addr) twice if the device for ucode is not found.

The problem was indicated by cppcheck.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 drivers/qe/qe.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/qe/qe.c b/drivers/qe/qe.c
index 1a829b5a4c..259e4e3e67 100644
--- a/drivers/qe/qe.c
+++ b/drivers/qe/qe.c
@@ -288,7 +288,6 @@ void u_qe_init(void)
 	struct mmc *mmc = find_mmc_device(CONFIG_SYS_MMC_ENV_DEV);
 
 	if (!mmc) {
-		free(addr);
 		printf("\nMMC cannot find device for ucode\n");
 	} else {
 		printf("\nMMC read: dev # %u, block # %u, count %u ...\n",
-- 
2.29.2

